homeassistant/components/ezviz/number.py

```diff
@@ -1,29 +1,59 @@
 """Support for EZVIZ number controls."""
 from __future__ import annotations
 
-from pyezviz.constants import DeviceCatagories
-from pyezviz.exceptions import HTTPError, PyEzvizError
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from pyezviz.constants import SupportExt
+from pyezviz.exceptions import (
+    EzvizAuthTokenExpired,
+    EzvizAuthVerificationCode,
+    HTTPError,
+    InvalidURL,
+    PyEzvizError,
+)
 
 from homeassistant.components.number import NumberEntity, NumberEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import EntityCategory
 from homeassistant.core import HomeAssistant
-from homeassistant.exceptions import HomeAssistantError
+from homeassistant.exceptions import ConfigEntryAuthFailed, HomeAssistantError
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .const import DATA_COORDINATOR, DOMAIN
 from .coordinator import EzvizDataUpdateCoordinator
-from .entity import EzvizEntity
+from .entity import EzvizBaseEntity
+
+SCAN_INTERVAL = timedelta(seconds=3600)
+PARALLEL_UPDATES = 0
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class EzvizNumberEntityDescriptionMixin:
+    """Mixin values for EZVIZ Number entities."""
+
+    supported_ext: str
+    supported_ext_value: list
+
 
-PARALLEL_UPDATES = 1
+@dataclass
+class EzvizNumberEntityDescription(
+    NumberEntityDescription, EzvizNumberEntityDescriptionMixin
+):
+    """Describe a EZVIZ Number."""
 
-NUMBER_TYPES = NumberEntityDescription(
+
+NUMBER_TYPES = EzvizNumberEntityDescription(
```

Comment:
Why is it named something in plural for a single description?

Reply:
Changed the name.